doc: clarify --watch and --watch-path usage with --run

juanibe opened the pull request:
This PR adds a note under the `--watch` and `--watch-path` CLI options in `cli.md` to clarify that these flags require a file path and cannot be used with `--run`.

Fixes confusion where users might assume `--watch-path` or `--watch` work with `--run`, which currently results in a runtime error.

Helps address issue discussion [here](https://github.com/nodejs/node/issues/58113)

shivarm commented on doc/api/cli.md:
```diff
@@ -3083,6 +3086,9 @@ combination with `--watch`.
 
+> **Note**: The `--watch-path` flag must be used alongside a script file. It cannot be used with `--run`, since watch mode requires a file to monitor and re-run. If used without a file, Node.js will throw an error.
+
+
```
remove an extra line

juanibe replied:
Done, thanks

juanarbol commented:
Hi! Thanks for taking care of this.

The PR may need some extra edits. The implies of watch in watch-path is not being shown, and also, just keep in mind that the --run flag has priority; It can not be mixed with watch and its derivatives. Could you please reflect that into your PR?

Refs:
https://github.com/nodejs/node/blob/main/src/node.cc#L1067

juanibe commented:
Hi @juanarbol!, thanks for the observation
I added the missing detail that --watch-path implicitly enables --watch and simplified the wording to reduce redundancy (since the reader can now refer to --watch). Let me know if any further clarification or modification is needed

juanarbol commented on doc/api/cli.md:
```diff
@@ -3083,6 +3085,8 @@ combination with `--watch`.
 This flag cannot be combined with
 `--check`, `--eval`, `--interactive`, `--test`, or the REPL.
 
+Note: Using `--watch-path` implicitly enables `--watch`, which requires a file path and is incompatible with `--run`, as `--run` takes precedence and disables watch mode.
```
```suggestion
Note: Using `--watch-path` implicitly enables `--watch`, which requires a file path and is incompatible with `--run`, as `--run` takes precedence and ignores watch mode.
```

juanarbol commented on doc/api/cli.md:
```diff
@@ -3056,6 +3056,8 @@ Use `--watch-path` to specify what paths to watch.
 This flag cannot be combined with
 `--check`, `--eval`, `--interactive`, or the REPL.
 
+Note: The `--watch` flag requires a file path as an argument. It does not support being used with `--run` or inline script input. If no file is provided, Node.js will throw an error.
```
I'm sorry, I forgot to mention that Node.js is not throwing, it will exit with status code 9. Won't throw.

juanibe replied:
Thanks for the correction, already changed that.

juanarbol reviewed (changes requested):
Just some more minor clarifications, I'm sorry I just forgot a couple details, but after this; I'm more than happy with this PR. Thanks again!

juanarbol reviewed (approved):
RLGTM

juanarbol commented:
Worth reading doc.

https://github.com/nodejs/node/blob/70863fb9dd4b70f1ba6141db4403e69eb7a9eeff/CONTRIBUTING.md :)

juanarbol reviewed (approved):
lgtm

juanarbol commented:
Test failure is already reported. 

Refs: https://github.com/nodejs/node/issues/58127

juanibe commented:
Updated the first commit message accordingly by amending it

juanarbol commented:
> Updated the first commit message accordingly by amending it

Thanks. **You don't have to do it now**, but if you intend to land a single commit, squash everything into a single commit.

juanibe commented:
> > Updated the first commit message accordingly by amending it
> 
> Thanks. **You don't have to do it now**, but if you intend to land a single commit, squash everything into a single commit.

Thanks! I've squashed the commits and the PR should be ready now. Let me know if there's anything else needed from my side.

juanarbol commented:
> Thanks! I've squashed the commits and the PR should be ready now. Let me know if there's anything else needed from my side.

I'm sorry, CI still not happy due to commit guidelines. Fix it, will approve and wait for more reviews :-)

Humble ping to @nodejs/documentation 

Refs:
https://github.com/nodejs/node/actions/runs/14824640752/job/41770212057#step:6:15

juanibe commented:
Yeah, missed that part 🙏.
Already amended it 
Thanks!

bnb commented on doc/api/cli.md:
```diff
@@ -3056,6 +3056,10 @@ Use `--watch-path` to specify what paths to watch.
 
+Note: The `--watch` flag requires a file path as an argument. It does not support
+being used with `--run` or inline script input. If no file is provided, Node.js will
+exit with status code 9.
```
```suggestion
exit with status code `9`.
```

bnb commented on doc/api/cli.md:
```diff
@@ -3056,6 +3056,10 @@ Use `--watch-path` to specify what paths to watch.
 `--check`, `--eval`, `--interactive`, or the REPL.
 
+Note: The `--watch` flag requires a file path as an argument. It does not support
+being used with `--run` or inline script input. If no file is provided, Node.js will
```
The second addition in this PR gives context as to why `--run` doesn't work, but this section doesn't. Ideally, both sections would give context.

juanibe replied:
Thanks for the suggestion! I originally didn’t include the extra context in the --watch section because I was inferring that, since --watch-path implicitly enables --watch, the reader could refer to the note below for more detail. But I understand how including it explicitly improves clarity and consistency — If this one is ok to you I can include and update the PR:

Note: The `--watch` flag requires a file path as an argument and is incompatible with `--run` or inline script input, as `--run` takes precedence and ignores watch mode. If no file is provided, Node.js will exit with status code `9`.

juanarbol commented:
@juanibe https://github.com/nodejs/node/actions/runs/14916227242/job/42382918668?pr=58136#step:7:21

Please, please, always run linters/tests before exposing for review. Once CI is happy, this will be landed.

juanibe commented:
> @juanibe https://github.com/nodejs/node/actions/runs/14916227242/job/42382918668?pr=58136#step:7:21
> 
> Please, please, always run linters/tests before exposing it for review. Once CI is happy, this will be landed.

Thanks @juanarbol for the reminder I've just run `make format-md` and pushed the changes. Let me know if there's anything else I should take care of.

juanarbol commented:
Landed in 0d3e1b3985ad 🎉
